packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -433,6 +467,165 @@ export class GraphQLApi extends Construct {
     });
   }
 
+  /**
+   * Adds an IAM policy statement associated with this GraphQLApi to an IAM
+   * principal's policy.
+   *
+   * @param grantee The principal (no-op if undefined)
+   * @param resources The set of resources to allow (i.e. ...:[region]:[accountId]:apis/GraphQLId/...)
+   * @param actions The actions that should be granted to the principal (i.e. appsync:graphql )
+   */
+  public grant(grantee: IGrantable, resources: IamResources[], ...actions: string[]): Grant {
+    return Grant.addToPrincipal({
+      grantee,
+      actions,
+      resourceArns: [...resources.map(resource => Lazy.stringValue({
+        produce: () => {
+          return this.generateResourceArn(resource);
+        },
+      }))],
+      scope: this,
+    });
+
+  }
+
+  /**
+   * Adds an IAM policy statement for full access to this GraphQLApi to an IAM
+   * principal's policy.
+   *
+   * @param grantee The principal (no-op if undefined)
+   */
+  public grantFullAccess(grantee: IGrantable): Grant {
+    return Grant.addToPrincipal({
+      grantee,
+      actions: ['appsync:graphql'],
+      resourceArns: [
+        this.generateResourceArn({custom: '*'}),
+      ],
+      scope: this,
+    });
+  }
+
+  /**
+   * Adds an IAM policy statement for Mutation access to this GraphQLApi to an IAM
+   * principal's policy.
+   *
+   * @param grantee The principal (no-op if undefined)
+   * @param fields The fields to grant access to that are Mutations (leave blank for all)
+   */
+  public grantMutation(grantee: IGrantable, fields?: [string]): Grant {
+    return this.grantType(grantee, 'Mutation', fields);
+  }
+
+  /**
+   * Adds an IAM policy statement for Query access to this GraphQLApi to an IAM
+   * principal's policy.
+   *
+   * @param grantee The principal (no-op if undefined)
+   * @param fields The fields to grant access to that are Queries (leave blank for all)
+   */
+  public grantQuery(grantee: IGrantable, fields?: [string]): Grant {
+    return this.grantType(grantee, 'Query', fields);
+  }
+
+  /**
+   * Adds an IAM policy statement for Subscription access to this GraphQLApi to an IAM
+   * principal's policy.
+   *
+   * @param grantee The principal (no-op if undefined)
+   * @param fields The fields to grant access to that are Subscriptions (leave blank for all)
+   */
+  public grantSubscription(grantee: IGrantable, fields?: [string]): Grant {
+    return this.grantType(grantee, 'Subscription', fields);
+  }
+
+  /**
+   * Adds an IAM policy statement for Type access to this GraphQLApi to an IAM
+   * principal's policy.
+   *
+   * @param grantee The principal (no-op if undefined)
+   * @param type The type to grant access
+   * @param fields The fields to grant access to that are of @param type (leave blank for all)
+   */
+  public grantType(grantee: IGrantable, type: string, fields?: [string]): Grant {
```

Comment:
Is there a reason to make this public?

Reply:
if a user makes their own type

```gql
type customType @aws_iam {
  ...
}
```

in order to do a mutation on that type, the `IAM` principal must have allowed permissions to the resourceArn